Intro/README.md

```diff
@@ -0,0 +1,455 @@
+   complex things like PKIX certificate validation path construction, or
+   even PKIX certificate path validation, nor do they implement complex
+   authentication protocols using PKIX, nor Kerberos, nor even simple
+   protocols like JWT.  Hardware TPMs can be used to implement portions
+   of such complex protocols (e.g., certificate issuance), but they
+   cannot implement the entirety of such protocols.
+
+ - Hardware TPMs are _slow_.
+
+ - A *lot* of detail is left out here.  Nothing is said about how TPMs
+   interface with the system or applications.  Little is said about
+   specific commands, and nothing is said about most commands.  Nothing
+   is said about TPM software stacks or APIs.
+
+   Our goal is only to help the lay reader understand TPMs well enough
+   at the conceptual level that they will be more comfortable
+   approaching mure more detailed TPM-related tutorials, documents,
+   books, or TCG specifications.
+
+## Hash Extension
+
+Hash extension is just appending some data to a current digest-sized
+value, hashing that, and then calling the output the new current value:
+
+```
+  v_0 = 0         # size-of-digest-output zero bits
+  v_1 = Extend(v_0, e_0)
+      = H(v_0 || e_0)
+  v_2 = Extend(v_1, e_1)
+      = H(v_1 || e_1)
+      = H(H(v_0 || e_0) || e_1)
+  v_3 = Extend(v_2, e_2)
+      = H(v_2 || e_2)
+      = H(H(v_1 || e_1) || e_2)
+      = H(H(H(v_0 || e_0) || e_1) || e_2)
+  ..
+  v_n = Extend(v_n-1, e_n-1)
+      = H(v_n-1 || e_n-1)
+      = H(H(v_n-2 || e_n-2) || e_n-1)
+      = H(H(...) || e_n-1)
+```
+
+where `H()` is a cryptographic hash function.
+
+Hash extension is most of what a PCR is, but hash extension is in other
+TPM concepts besides PCRs, such as policy naming.
+
+## Platform Configuration Registers (PCRs)
+
+A PCR, then, is just a hash output.  The only operations on PCRs are:
+read, extend, and reset.  All richness of semantics of PCRs come from
+how they are used:
+
+ - how they are extended and by what code
+ - what purposes they are read for
+    - attestation
+    - authorization
+
+## Root of Trust Measurements (RTM)
+
+When a computer and its TPM start up, most PCRs are set to all-zeros,
+and then the computer's boot firmware performs a core root of trust
+measurement (CRTM) to "measure" (i.e., hash) the the next boot stage and
+extend it into an agreed-upon PCR.  The entire boot process should,
+ideally, perform RTMs.  At the end of the boot process some set of PCRs
+should reflect the totality of the code path taken to complete booting.
+
+Some PCRs are used to measure the BIOS, others to measure option ROMs,
+and others to measure the operating system.  In principle one could
+measure the entirety of an operating system and all the code that is
+installed on the system.
+
+RTM can be used to ensure that only known-trusted code is executed, and
+that important resources are not unlocked unless the state of the system
+when they are needed is "only executed trusted code".
```

Comment:
Maybe "the system has only executed trusted code"? Just "only executed trusted code" isn't super clear to me

Reply:
I'll think about re-phrasing this.